pyccel/ast/operators.py

```diff
@@ -85,7 +85,7 @@ def broadcast(shape_1, shape_2):
         else:
             msg = 'operands could not be broadcast together with shapes {} {}'
             msg = msg.format(shape_1, shape_2)
-            errors.report(msg,severity='fatal')
+            raise PyccelSemanticError(msg)
```

Comment:
Why do you need to raise an Exception instead of using `Errors.report`?

Reply:
The error is caught [here](https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L1124) so that the context information can be added (symbol information). If we use `errors.report` to raise that error then we get the same message in the output twice.
E.g.:
```
ERROR at annotation (semantic) stage
pyccel:
 |fatal [semantic]: test.py| operands could not be broadcast together with shapes (Shape(p,0) - 3,) (Shape(p,0) - 2,)
 |fatal [semantic]: test.py [6,4]| operands could not be broadcast together with shapes (Shape(p,0) - 3,) (Shape(p,0) - 2,) (p[3 : ] + p[2 : ])
```

Reply:
Nice, thanks!